### PR TITLE
Director service sequence diagrams

### PR DESCRIPTION
**Pull request by igor-davidyuk**
Rendered version: https://openfl.readthedocs.io/en/mermaid_sequence_diagrams/source/openfl/communication.html#

### CODE REVIEW

File: docs/source/openfl/director_envoy.mmd [hunk 1]
```diff
@@ -0,0 +1,30 @@
+sequenceDiagram
+    participant D as Director
+    participant E as Envoy
+    rect rgb(0, 255, 0,.1)
+        Note over D,E: A Federation startup process
+        D->D: Starts
+        E->E: Starts and loads local Shard Descriptor
+        E-->>D: Connects using FQDN and pwd
```
Comment (dmitryagapov) [inline docs/source/openfl/director_envoy.mmd]: pwd == certificates?

File: docs/source/openfl/director_envoy.mmd [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        E-->>+D: Communicates dataset info
+        D-->D: Ensures unified data interface
+        D-->>-E: Approves
+        D-->D: Keeps a list of connected Envoys
```
Comment (dmitryagapov) [inline docs/source/openfl/director_envoy.mmd:12]: Maybe "Add to connected Envoys list"

File: docs/source/openfl/director_envoy.mmd [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        
+    end
+    Note over D,E: We consider a Federation set up
+    rect rgb(0, 0, 255,.05)
+        Note over D,E: An Experiment's start
```
Comment (dmitryagapov) [inline docs/source/openfl/director_envoy.mmd:17]: Maybe first step should be "Add experiment to Experiments queue"

File: docs/source/openfl/director_envoy.mmd [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        opt Test run failed
+            D-->D: Notifies user that <br>the experiment is inconsistent
+            D-->D: The experiment ends
+        end
+        D->D: Starts an Aggregator
+        D-->>E: Sends the experiment archive <br>and the FL plan to envoys
+        E-->E: Starts a Collaborator
+        D-->D: Fills the model with the final weights <br>and returns to user
+    end
+    Note over D,E: The Experiment ended. <br> The Federation keeps existing.
```
Comment (dmitryagapov) [inline docs/source/openfl/director_envoy.mmd:30]: Maybe "D-->D: Fills the model with the final weights <br>and returns to user" should be only "D-->D: Fills the model with the final weights"
and
after "Note over D,E: The Experiment ended. <br> The Federation keeps existing."
add "Note over D,E: User get model from Director"

File: docs/source/openfl/experiment_representation_and_RPCs.mmd [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    D->>-F: Response with model structure and the plan
+  end
+
+
+  F->>+D: Requests training artifacts
+  D->>ER: Gets training artifacts
+  D->>-F: Respond
+
+  F->>+D: Requests removing the Experiment entry
+  D->>ER: Removes Experiment from registry
```
Comment (dmitryagapov) [inline docs/source/openfl/experiment_representation_and_RPCs.mmd:53]: Maybe we need to add 
D->>-f: Response/Approve

Comment (igor-davidyuk) [reply docs/source/openfl/experiment_representation_and_RPCs.mmd:53]: Sure, done

File: docs/source/openfl/director_envoy.mmd (hunk 1 above)
Comment (igor-davidyuk) [reply docs/source/openfl/director_envoy.mmd]: Correct, thanks

**Pull request merged**